pkgs/applications/misc/intune-portal/default.nix

```diff
@@ -0,0 +1,83 @@
+{ stdenv
+, lib
+, fetchurl
+, dpkg
+, libuuid
+, xorg
+, curlMinimal
+, openssl
+, libsecret
+, webkitgtk
+, libsoup
+, gtk3
+, atk
+, pango
+, glib
+, sqlite
+, zlib
+, systemd
+, msalsdk-dbusclient
+, pam
+}:
+stdenv.mkDerivation rec {
+  pname = "intune-portal";
+  version = "1.2303.10";
+
+  src = fetchurl {
+    url = "https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/i/${pname}/${pname}_${version}_amd64.deb";
+    hash = "sha512-ikaZSzN/1VBMIKTHUytisGk4JxIwzLMCZRchTlKt1dOotJExLbYmNFwckxmYEKRHjZLDPUKOumcI8bwavJyadA==";
+  };
+
+  nativeBuildInputs = [ dpkg ];
+
+  unpackCmd = ''
+    mkdir -p root
+    dpkg-deb -x $curSrc root
+  '';
+
+
+  buildPhase =
+    let
+      libPath = {
+        intune = lib.makeLibraryPath [ stdenv.cc.cc.lib libuuid xorg.libX11 curlMinimal openssl libsecret webkitgtk libsoup gtk3 atk glib pango sqlite zlib systemd msalsdk-dbusclient ];
+        pam = lib.makeLibraryPath [ pam ];
+      };
+    in
+    ''
+      patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) --set-rpath ${libPath.intune} opt/microsoft/intune/bin/intune-portal
+      patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) --set-rpath ${libPath.intune} opt/microsoft/intune/bin/intune-agent
+      patchelf --set-rpath ${libPath.pam} ./usr/lib/x86_64-linux-gnu/security/pam_intune.so
+    '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    cp -a opt/microsoft/intune/bin/* $out/bin/
+    cp -a usr/share $out
+    cp -a lib $out
+    mkdir -p $out/lib/security
+    cp -a ./usr/lib/x86_64-linux-gnu/security/pam_intune.so $out/lib/security/
+    cp -a usr/lib/tmpfiles.d $out/lib
+
+    substituteInPlace $out/share/applications/intune-portal.desktop \
+      --replace /opt/microsoft/intune/bin/intune-portal $out/bin/intune-portal
+
+    substituteInPlace $out/lib/systemd/user/intune-agent.service \
+      --replace \
+        ExecStart=/opt/microsoft/intune/bin/intune-agent \
+        ExecStart=$out/bin/intune-agent
+
+    runHook postInstall
+  '';
+
+  dontPatchELF = true;
```

Comment:
Why does the normal patchelf not work?

Reply:
I'm yet to figure out why. But when enabled the produced binaries network connections don't work properly.

Reply:
@rhysmdnz any pointer on how you debugged this network issue? 
I'm trying to use the service but I'm getting (Code:2604) Network infrastructure failure. I've tried building and using intune-portal in the version in which it was initialized but no luck.